overlord/snapstate/policy/snapd.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package policy
+
+import (
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+type snapdPolicy struct {
+	modelBase string
+}
+
+func (p *snapdPolicy) CanRemove(st *state.State, snapst *snapstate.SnapState, rev snap.Revision) error {
+	name := snapst.InstanceName()
+	if name == "" {
+		// not installed, or something. What are you even trying to do.
+		return errNoName
+	}
+
+	if !rev.Unset() {
+		return nil
+	}
+
+	// snapd cannot be removed on core
+	if !release.OnClassic {
```

Comment:
hmm, wouldn't using `model.Classic()` make more sense here?

Reply:
Maybe, it should be equivalent - or am I missing something?

Reply:
I think it fits better with policy if you grab classic from the model, as we do with base in other policies.
I'm not sure there's a stronger reason than that, as I don't think you can remodel from classic to non-, but the fact that I need to ask myself this points you at why it's saner to get it from the model that was passed in...